develop/devguide/Automation/Howto/Creating_a_cypress_test_for_an_interactive_automation_script.md

```diff
@@ -0,0 +1,219 @@
+## Setting up the web environment
+
+1. [Create a new low-code app](xref:Creating_custom_apps).
+
+1. Add a button to the low-code app and configure it to trigger the created interactive Automation script when clicked.
+
+   For more information, see [Button](xref:DashboardButton).
+
+## Creating the Cypress test
+
+1. In the `cypress/e2e` folder, create a test file with a `.cy.js` extension, such as `HelloWorldDialog.cy.js`.
+
+1. Implement the Cypress test below to log in, navigate to the low-code app, and interact with the Automation script.
+
+   > [!NOTE]
+   > Ensure all variable values in the test below are updated with the correct information (*dmaAddress*, *username*, *password*, etc.) before you run the test. This allows the script to interact properly with your specific application environment.
+
+   ```javascript
+   describe('Test Interactive Automation Script', () => {
+       // Define the necessary variables and replace with your specific details
+       const dmaAddress = ''; // Replace with the IP or hostname of a DMA (e.g. 'localhost')
+       const username = ''; // Replace with the user username
+       const password = ''; // Replace with the user password
+       const buttonLabel = ''; // Replace with the button label
+       const appId = ''; // Replace with the app ID
+
+       // Before running the tests, visit the low-code app
+       before(() => cy.visit(`http://${dmaAddress}/app/${appId}`));
+
+       it('should log in and execute the Interactive Automation Script', () => {
+           // Check if the login component exists
+           cy.get('dma-login').should('exist');
+
+           // Enter login credentials
+           cy.get('[data-cy="login.username"]').type(username);
+           cy.get('[data-cy="login.password"]').type(password);
+
+           // Click the login button
+           cy.contains('dma-button', 'Log on').click();
+
+           // Verify successful login
+           cy.get('dma-login').should('not.exist');
+
+           // Start the IAS by clicking the associated button
+           cy.contains('dma-button', buttonLabel).click();
+
+           // Validate that the pop-up message appears with the correct title
+           cy.get('[data-cy="popup"]').contains('Dialog').should('be.visible');
+
+           // Get the component with the debug tag 'label' and check if the component has the correct text
+           cy.get('[data-cy="label"]').should('have.text', 'Hello, World!');
+
+           // Get the component with the debug tag 'button' and click the OK button
+           cy.get('[data-cy="button"]').should('have.text', 'OK').click();
+       });
+   });
+   ```
+
+## Running the Cypress test
+
+1. Launch Cypress by running the following command in the terminal:
+
+   ```cmd
+   npm run start
+   ```
+
+1. Select the test file (`HelloWorldDialog.cy.ts`).
+
+   Cypress will execute the test, displaying results in real time: green indicates success, and any errors are logged in detail for debugging.
```

Comment:
S: Add an extra step to make the test fail. For example: in the Interactive Automation Script change the text of the label from "Hello, World!" to "Goodbye, World!".

![image](https://github.com/user-attachments/assets/0f2854ec-1259-4757-a0e1-58774033d452)